src/Pin/index.js

```diff
@@ -23,21 +23,25 @@ import { ApplicationInstance } from '../Launch'
 
 // only used during local development
 let unlocked = false
+let contextItems = ['purchase', 'parental']
 
-let submit = pin => {
+let submit = (pin, context) => {
   return new Promise((resolve, reject) => {
-    if (pin.toString() === Settings.get('platform', 'pin', '0000').toString()) {
+    if (
+      pin.toString() === Settings.get('platform', 'pin', '0000').toString() &&
+      contextItems.includes(context)
+    ) {
```

Comment:
We shouldn't perform the valid `context` check here, since this function is overwritten by the platform plugin. Instead it should be added to the public API `submit`-method, before we pass it to the actual `submit` method.

This will allow to throw a specific log about the invalid context. Context should still be passed to the platform submit method though